mlflow/tracking/fluent.py

```diff
@@ -69,6 +70,7 @@
     import plotly
 
 _active_run_stack = []
+run_id_to_system_metrics_monitor = {}
```

Comment:
i'm not sure how common this is at all, but perhaps inverting this into a singleton monitor, which can report metrics to N runs would be a better approach?

Reply:
thanks for the suggestion, I think essentially that's a similar approach? The gist of singleton or global variable is we maintain only one instance during the program's lifecycle, just a different way on expressing it. In the current codebase, we are using global variables (like the above `_active_run_stack `) more often than singletons, so I would prefer to use a consistent style.